src/components/Tooltip/index.tsx

```diff
@@ -1,58 +1,73 @@
+  arrowOffset: number
+}
+
 const TooltipPanel = styled.div`
-  width: 300px;
-  height: auto;
-  padding: 15px;
+  width: ${(props: TooltipPanelProps) => props.width};
+  padding: 20px 30px 20px 30px;
   background-color: #676767;
-  border-bottom-color: #676767;
+  border-radius: 6px;
+  box-shadow: 2px 2px 6px 0 #b3b3b3;
+  word-break: break-word;
   color: #ffffff;
   font-size: 13px;
-  font-family: PingFangSC-Semibold;
-  line-height: 18px;
-  box-sizing: border-box;
-  border-radius: 6px;
+  left: 0px;
+  transform: translate(
+    ${(props: TooltipPanelProps) => (props.offset ? `${props.offset.x}px` : '0px')},
+    ${(props: TooltipPanelProps) => (props.offset ? `${props.offset.y}px` : '10px')}
+  );
   position: absolute;
-  word-break: break-word;
   z-index: 2;
-  top: ${(props: { targetSize: TargetSize }) => props.targetSize.height}px;
-  left: ${(props: { targetSize: TargetSize }) => (props.targetSize.width - 300) * 0.5}px;
 
   @media (max-width: 700px) {
-    width: 200px;
-    font-size: 12px;
-    left: ${(props: { targetSize: TargetSize }) =>
-      (props.targetSize.width - 200) * (props.targetSize.offset ? props.targetSize.offset + 0.04 : 0.5)}px;
+    font-size: 11px;
+    padding: 10px 15px 10px 15px;
   }
 
-  &:: after {
+  &::after {
     content: '';
-    width: 10px;
-    height: 10px;
-    background-color: inherit;
-    top: -5px;
-    left: calc(50% - 5px);
-
-    @media (max-width: 700px) {
-      left: calc(
-        ${(props: { targetSize: TargetSize }) =>
-            props.targetSize.offset ? `${props.targetSize.offset * 100}%` : '50%'} - 5px
-      );
-    }
+    left: ${(props: TooltipPanelProps) => `${props.arrowOffset - 16}px`};
     position: absolute;
-    transform: rotate(45deg);
+    top: -6px;
+    border: 16px solid transparent;
+    border-bottom: 10px solid #676767;
+    border-top: 0;
   }
 `
 
-export interface TargetSize {
-  width: number
-  height: number
-  offset?: number
+interface TooltipProps {
+  show: boolean
+  targetElementId: string
+  width?: string
+  offset?: { x: number; y: number }
+  children?: ReactNode
 }
 
-const Tooltip = ({ show = false, message, targetSize }: { show: boolean; message: string; targetSize: TargetSize }) => {
-  return <React.Fragment>{show && <TooltipPanel targetSize={targetSize}>{message}</TooltipPanel>}</React.Fragment>
+export default ({ show, targetElementId, width = '100%', offset, children }: TooltipProps) => {
+  const [arrowOffset, setArrowOffset] = useState(0)
+  useLayoutEffect(() => {
+    const currentElement = document.getElementById('TooltipPanel')
+    if (currentElement) {
+      const targetElement = document.getElementById(targetElementId)
+      if (targetElement) {
+        const currentReact = currentElement.getBoundingClientRect()
+        const targetReact = targetElement.getBoundingClientRect()
+        setArrowOffset(targetReact.left - currentReact.left + targetReact.width / 2)
+      }
+    }
+  }, [show, targetElementId])
```

Comment:
`useLayoutEffect` dose not have `useEffect/useMemo/useCallback`, so what does dependences mean ?

Reply:
If don't add a dependency, it will only be executed once, and it will not be executed again when `show` changes.

Reply:
OK. I misunderstood